ff/serving/coordinator/coordinator.go

```diff
@@ -90,19 +107,96 @@ func (c *Coordinator) WatchForNewJobs() error {
 	return nil
 }
 
-func (c *Coordinator) runRegisterSourceJob(resID metadata.ResourceID) error {
-	//the idea is that we register a csv or a postgres table
-	//and via that provider we should be able to "get" any of those offline tables from here on out
-	//and if a feature declares a source in its definition, the training set job should be able to go
-	//and get the namevariant shit
+func (c *Coordinator) mapNameVariantsToTables(sources []metadata.NameVariant) (map[string]string, error) {
+	sourceMap := make(map[string]string)
+	for _, nameVariant := range sources {
+		var tableName string
+		source, err := c.Metadata.GetSourceVariant(context.Background(), nameVariant)
+		if err != nil {
+			return nil, err
+		}
+		if source.Status() != metadata.READY {
+			return nil, fmt.Errorf("source in query not ready")
+		}
+		providerResourceID := provider.ResourceID{Name: source.Name(), Variant: source.Variant()}
+		if source.IsSQLTransformation() {
+			tableName = provider.GetTransformationName(providerResourceID)
+		} else if source.IsPrimaryDataSQLTable() {
+			tableName = provider.GetPrimaryTableName(providerResourceID)
+		}
+		sourceMap[fmt.Sprintf("%s.%s", nameVariant.Name, nameVariant.Variant)] = tableName
+	}
+	return sourceMap, nil
+}
+
+func sanitize(ident string) string {
+	return db.Identifier{ident}.Sanitize()
+}
+
+func (c *Coordinator) runSQLTransformationJob(transformSource *metadata.SourceVariant, resID metadata.ResourceID, offlineStore provider.OfflineStore) error {
+	templateString := transformSource.SQLTransformationQuery()
+	sources := transformSource.SQLTransformationSources()
+	sourceMap, err := c.mapNameVariantsToTables(sources)
+	if err != nil {
+		return err
+	}
+	query := templateReplace(templateString, sourceMap)
+	providerResourceID := provider.ResourceID{Name: resID.Name, Variant: resID.Variant, Type: provider.Transformation}
+	transformationConfig := provider.TransformationConfig{TargetTableID: providerResourceID, Query: query}
+	if err := offlineStore.CreateTransformation(transformationConfig); err != nil {
+		return err
+	}
+	if err := c.Metadata.SetStatus(context.Background(), resID, metadata.READY); err != nil {
+		return err
+	}
 	return nil
+	//do we set this in metadata?
+	//GetTransformationTable(id ResourceID) (TransformationTable, error)
+
 }
 
-func (c *Coordinator) runTransformationJob(resID metadata.ResourceID) error {
-	//it's morphin time
+func (c *Coordinator) runPrimaryTableJob(transformSource *metadata.SourceVariant, resID metadata.ResourceID, offlineStore provider.OfflineStore) error {
+	providerResourceID := provider.ResourceID{Name: resID.Name, Variant: resID.Variant}
+	sourceName := transformSource.PrimaryDataSQLTableName()
+	if sourceName == "" {
+		return fmt.Errorf("creating blank source not implemented")
```

Comment:
Would we ever implement it? If not just say its an error and not that its unimplemented

Reply:
We will I imagine, creating an empty primary table with a predefined schema.